src/Composer/Package/Archiver/ArchiveManager.php

```diff
@@ -71,36 +71,61 @@ public function setOverwriteFiles(bool $overwriteFiles): self
     }
 
     /**
-     * Generate a distinct filename for a particular version of a package.
-     *
-     * @param CompletePackageInterface $package The package to get a name for
-     *
-     * @return string A filename without an extension
+     * @return array<string, string>
```

Comment:
Let's avoid expanding the public method surface here

```suggestion
     * @return array<string, string>
     * @internal
```